connect-location/src/test/java/com/ifttt/location/ConnectLocationTest.java

```diff
@@ -0,0 +1,35 @@
+package com.ifttt.location;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import com.ifttt.connect.ui.ConnectButton;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.annotation.Config;
+import org.robolectric.annotation.LooperMode;
+
+import static org.junit.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(sdk = 28)
+public class ConnectLocationTest {
+
+    private ConnectButton button;
+
+    @Before
+    public void setUp(){
+        ActivityScenario<TestActivity> scenario = ActivityScenario.launch(TestActivity.class);
+        scenario.onActivity(activity -> {
+            button = activity.findViewById(R.id.ifttt_connect_button_test);
+        });
+    }
+
+
+    @Test(expected = IllegalStateException.class)
+    public void setUpWithoutInit() {
+        ConnectLocation.setUpWithConnectButton(button);
```

Comment:
Not sure if we need the TestActivity to test this method. We can use just the `Activity` to create the ActivityScenario.

Reply:
we want to initialize the `ConnectButton`. The `TestActivity` assigns an id to it which we look up here to find the view.

Reply:
Is it not possible to use `new ConnectButton(activity)`?

Reply:
It is, but I believe it is not a good practice to initialize a view class that way, since it extends `FrameLayout`. But again we are doing that for the test activity anyways. I will remove `TestActivity`

Reply:
I think it depends, there are times when using xml is not applicable (like when we need to render something dynamically), there is nothing special about initializing with xml or directly using the constructor, it's just using the default attributes and styles.

Reply:
talked irl and we are keeping the `TestActivity` since `ActivityScenario` cannot work with generic Activity class.